Mage/src/main/java/mage/abilities/dynamicvalue/common/TotalPermanentsManaValue.java

```diff
@@ -0,0 +1,58 @@
+package mage.abilities.dynamicvalue.common;
+
+import mage.abilities.Ability;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.effects.Effect;
+import mage.abilities.hint.Hint;
+import mage.abilities.hint.ValueHint;
+import mage.filter.FilterPermanent;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+
+import java.util.List;
+
+public class TotalPermanentsManaValue implements DynamicValue {
+
+    private final String message;
+    private final Hint hint;
+    private final FilterPermanent filter;
+
+    public TotalPermanentsManaValue(FilterPermanent filter) {
+        this.filter = filter.copy();
+        this.message = "the total mana value of " + filter.getMessage();
+        this.hint = new ValueHint("Total mana value of " + filter.getMessage(), this);
+    }
+
+    @Override
+    public DynamicValue copy() {
```

Comment:
Are you sure this is a sufficient copy method? I'm not super familiar but there is this comment:

https://github.com/magefree/mage/blob/9eb7e2870e6536152e94c8c886b53c688b6f1aa6/Mage/src/main/java/mage/abilities/dynamicvalue/DynamicValue.java#L10-L14

Reply:
I can make a regular copy method instead, but `DevotionCount` was doing it that way.

Reply:
DevotionCount is an enum, does that make a difference? I am not sure, sorry

Reply:
I am not sure either.
I'll make the full copy / private constructor just to be safe.